### ISSUE TITLE
SSH freezes

### ISSUE THREAD

**Issue by bibinphilip** [hashicorp/vagrant#11741]
Hi,

### Vagrant version

Vagrant version 2.2.9

### Host operating system
Windows 10 Pro

### Provider
Hyper-V

### Guest operating system

Alpine

### Vagrantfile

[Vagrantfile.txt](https://github.com/hashicorp/vagrant/files/4872267/Vagrantfile.txt)

### Debug output

[Vagrant ssh log.txt](https://github.com/hashicorp/vagrant/files/4872265/Vagrant.ssh.log.txt)

### Expected behavior

Authenticate guest machine via SSH

### Actual behavior
Authentication of guest machine via SSH freezes during 'SSH auth method: private key'

When installing a new Alpine OS via "vagrant up" command, there is no further action during the SSH authentication while 'SSH auth method: private key' processes.

Please find attached debug output of the command "vagrant ssh..."

**Comment by briancain** [comment]
Hey there @bibinphilip - It looks like maybe the entire debug log was not included? I don't see any timeout information in the log provided.

Also, when you bring up a guest in Hyper-V, while trying to ssh, are you able to actually log into it through the Hyper-V manager gui? I noticed your Vagrantfile only gives the guest 1 CPU and 1 GB of RAM. I wouldn't be surprised if the guest just actually can't boot up due to the small amount of resources. And checking the GUI would give you an idea of what's happening with the guest.

**Comment by bibinphilip** [comment]
Hi @briancain ,
Thank you for your reply.

I am able to connect the VM via the GUI and able  to execute the normal LINUX commands.

I implemented your suggestion by increasing the resources, this time with Debian OS, but the result is same. The execution is going in loop "Command not in installer, restoring original environment", which I don't understand.

I have attached new log and vagrant file. Please share your thoughts and suggestions. 

[Vagrantfile.txt](https://github.com/hashicorp/vagrant/files/4882464/Vagrantfile.txt)
[vagrant up log.txt](https://github.com/hashicorp/vagrant/files/4882465/vagrant.up.log.txt)

Thank you,
Bibin Philip

**Comment by briancain** [comment]
That string about "command not in installer" isn't an error, it's just some debug output and can safely be ignored.

I don't see any errors in that log, it just looks like Vagrant was interrupted with `ctrl+c` before it was finished executing. But also I would still give that vm more than 1 GB of ram....and it's usually preferable to give it ~2 CPUs rather than 3 or 4. I'd try that, and let Vagrant run its course before interrupting it and see what happens.

**Comment by bibinphilip** [comment]
Yes @briancain , I interrupted the execution as there is no end. It will continue even after 5 minutes... :) I tried with your suggestion and provided 3GB Ram, 3 CPUs but still the same. Is this due to External Switch having Wi-fi adapter?

**Comment by briancain** [comment]
That's possible @bibinphilip . If you remove the public network, does the timeout not occur?

**Comment by bibinphilip** [comment]
Hi @briancain ,
I further checked with the following switches and then they are working fine:
* Default Switch
* External Switch with Ethernet adapter

It seems the issue is with External Switch with WiFi adapter, as they create a network bridge between the WiFi & the virtual switch. Hope this helps. Please suggest, if you have any solution for this. Thank you.

**Comment by briancain** [comment]
@bibinphilip - It's possible that using an external switch with your wifi adapter just doesn't work and your VM doesn't have proper networking. If you set up a vm outside of Vagrant through the Hyper-V management GUI and use the wifi adapter, does it result in the same issue with being unable to ssh?

**Comment by bibinphilip** [comment]
Thanks @briancain  for the suggestion, I need to check that. Would it be possible for you to share some links related to the issue of  network bridge with External Switch having WiFi adapter? I am not getting what is causing the issue... Thanks.
